src/NHibernate/Loader/Loader.cs

```diff
@@ -944,25 +944,59 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 
 			object[] rowResults = new object[cols];
 
+			var entityCounter = new Dictionary<string, int>();
+
 			for (int i = 0; i < cols; i++)
 			{
 				object obj = null;
+				ILoadable persister = persisters[i];
+				
+				entityCounter.TryGetValue(persister.EntityName, out int total);
+				entityCounter[persister.EntityName] = ++total;
+
 				EntityKey key = keys[i];
 
 				if (key == null)
 				{
-					// do nothing
-					/* TODO NH-1001 : if (persisters[i]...EntityType) is an OneToMany or a ManyToOne and
-					 * the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
-					 * an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
-					*/
+					if (cols > 1)
+					{
+						/*
+						* the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
+						* an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
+						*/
+						for (int j = 0; j < persisters.Length; j++)
+						{
+							if (i == j)
+							{
+								continue;
+							}
+
+							bool found = false;
+							int count = 0;
+							for (int z = 0; z < persisters[j].PropertyTypes.Length; z++)
+							{
+								IType type = persisters[j].PropertyTypes[z];
+								if (type.Name == persister.EntityName && ++count == total
```

Comment:
I'm afraid but we can't detect owner by order of persisters in result as order can be changed in query.
Here is the test that breaks this check:
```C#
[Test]
public void Department2IsNull_QueryOver()
{
	ExecuteStatement($"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 99999 WHERE EMPLOYEE_ID = {employeeId}");

	IStatistics statistics = Sfi.Statistics;
	statistics.Clear();

	using (ISession session = OpenSession())
	{
		var employee = session.QueryOver<Employee>()
			.Fetch(SelectMode.Fetch, e => e.Department2)
			.Fetch(SelectMode.Skip, e => e.Department1, e => e.Department3, e => e.Address)
			.Where(e => e.Id == employeeId)
			.SingleOrDefault();

		Assert.That(employee.Department1, Is.Not.Null);
		Assert.That(employee.Department2, Is.Null);
		Assert.That(employee.Department3, Is.Not.Null);
		Assert.That(employee.Address, Is.Not.Null);
		Assert.That(statistics.PrepareStatementCount, Is.EqualTo(1));
	}
}
```
We need some other way to properly find owner property.